pkg/envoy/rds/route/route_config.go

```diff
@@ -209,52 +211,52 @@ func buildEgressRoutes(routingRules []*trafficpolicy.EgressHTTPRoutingRule) []*x
 		// Build the route for the given egress routing rule and method
 		// Each HTTP method corresponds to a separate route
 		for _, httpMethod := range allowedHTTPMethods {
-			route := buildRoute(rule.Route.HTTPRouteMatch.PathMatchType, rule.Route.HTTPRouteMatch.Path, httpMethod, nil, rule.Route.WeightedClusters, rule.Route.RetryPolicy)
+			route := buildRoute(rule.Route, httpMethod)
 			routes = append(routes, route)
 		}
 	}
 	return routes
 }
 
-func buildRoute(pathMatchTypeType trafficpolicy.PathMatchType, path string, method string, headersMap map[string]string, weightedClusters mapset.Set, retryPolicy trafficpolicy.RetryPolicy) *xds_route.Route {
+func buildRoute(httpRoute trafficpolicy.RouteWeightedClusters, method string) *xds_route.Route {
```

Comment:
It might make sense to rename `httpRoute` var to something like `weightedClusters`.